crates/miden-lib/src/account/wallets/mod.rs

```diff
@@ -30,6 +31,47 @@ use crate::account::{auth::RpoFalcon512, components::basic_wallet_library};
 /// [kasm]: crate::transaction::TransactionKernel::assembler
 pub struct BasicWallet;
 
+impl BasicWallet {
+    // CONSTANTS
+    // --------------------------------------------------------------------------------------------
+    const RECEIVE_ASSET_PROC_NAME: &str = "receive_asset";
+    const CREATE_NOTE_PROC_NAME: &str = "create_note";
+    const MOVE_ASSET_TO_NOTE_PROC_NAME: &str = "move_asset_to_note";
+
+    // PUBLIC ACCESSORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns the digest of the `receive_asset` wallet procedure.
+    pub fn receive_asset_digest() -> Digest {
+        Self::get_procedure_digest_by_name(Self::RECEIVE_ASSET_PROC_NAME)
+    }
+
+    /// Returns the digest of the `create_note` wallet procedure.
+    pub fn create_note_digest() -> Digest {
+        Self::get_procedure_digest_by_name(Self::CREATE_NOTE_PROC_NAME)
+    }
+
+    /// Returns the digest of the `move_asset_to_note` wallet procedure.
+    pub fn move_asset_to_note_digest() -> Digest {
+        Self::get_procedure_digest_by_name(Self::MOVE_ASSET_TO_NOTE_PROC_NAME)
+    }
+
+    // HELPER FUNCTIONS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns the digest of the basic wallet procedure with the specified name.
+    fn get_procedure_digest_by_name(procedure_name: &str) -> Digest {
+        let proc_name = ProcedureName::new(procedure_name).expect("procedure name should be valid");
+        let module = basic_wallet_library()
+            .module_infos()
+            .next()
+            .expect("basic_wallet_library should have exactly one module");
+        module.get_procedure_digest_by_name(&proc_name).unwrap_or_else(|| {
+            panic!("basic_wallet_library should contain the '{proc_name}' procedure")
+        })
+    }
```

Comment:
Instead of doing this search on every invocation of `get_procedure_digest_by_name()`, is there a way to do this statically (e.g., using `LazyLock`)?

Also, the complexity of this makes me think that we are missing some convenience methods on the `Library` struct. For example, would be great to have something like:
```rust
impl Library {

    pub fn get_procedure_root_by_name(&self, proc_name: QualifiedProcedureName) -> Option<Word> {
        ...
    }
}
```
Could you create an issue for this in `miden-vm` (and we could make a small PR for this too).